test/extended/builds/pipeline_origin_bld.go

```diff
@@ -215,430 +86,34 @@ var _ = g.Describe("[sig-builds][Feature:Builds][sig-devex][Feature:Jenkins][Slo
-
-					g.By("verify credential deleted when secret deleted")
-					err = oc.Run("delete").Args("secret", secretName).Execute()
-					o.Expect(err).NotTo(o.HaveOccurred())
-					_, err = j.WaitForContent("", 404, 10*time.Second, "credentials/store/system/domain/_/credential/%s-%s/", oc.Namespace(), secretName)
-					if err != nil {
-						exutil.DumpApplicationPodLogs("jenkins", oc)
-					}
-					o.Expect(err).NotTo(o.HaveOccurred())
-
-					// no need to clean up, last operation above deleted the secret
-				})
-
-				//TODO - for these config map slave tests, as well and the imagestream/imagestreamtag
-				// tests ... rather than actually running the pipelines, we could just inspect the config in jenkins
-				// to make sure the k8s pod templates are there.
-				// In general, while we want at least one verification somewhere in pipeline.go that the agent
-				// images work, we should minimize the actually running of pipelines using them to only one
-				// for each maven/nodejs
-				g.By("Pipeline using config map slave")
+		g.Describe("jenkins pipeline build config strategy", func() {
+			g.It("using a jenkins instance launched with the ephemeral template", func() {
+				defer cleanup()
+				setupJenkins()
 
 				g.By("should build and complete successfully", func() {
-					g.By("create the pod template with config map")
-					err := oc.Run("create").Args("-f", configMapPodTemplatePath).Execute()
-					o.Expect(err).NotTo(o.HaveOccurred())
 
-					g.By(fmt.Sprintf("calling oc new-app -f %q", podTemplateSlavePipelinePath))
-					err = oc.Run("new-app").Args("-f", podTemplateSlavePipelinePath).Execute()
+					g.By("calling oc create -f to create buildconfig")
+					err := oc.Run("create").Args("-f", simplePipelineBC).Execute()
 					o.Expect(err).NotTo(o.HaveOccurred())
 
 					g.By("starting the pipeline build and waiting for it to complete")
 					// this just does sh "mvn --version"
-					br, err := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+					br, err := exutil.StartBuildAndWait(oc, "minimalpipeline")
 					if err != nil || !br.BuildSuccess {
-						debugAnyJenkinsFailure(br, oc.Namespace()+"-openshift-jee-sample", oc, true)
+						debugAnyJenkinsFailure(br, oc.Namespace(), oc, true)
 						exutil.DumpBuilds(oc)
 					}
 					br.AssertSuccess()
 
 					g.By("getting job log, make sure has success message")
-					out, err := j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
-					o.Expect(err).NotTo(o.HaveOccurred())
-					g.By("making sure job log ran with our config map slave pod template")
-					o.Expect(out).To(o.ContainSubstring("Running on jenkins-slave"))
-
-					g.By("clean up openshift resources for next potential run")
-					err = oc.Run("delete").Args("configmap", "jenkins-slave").Execute()
-					o.Expect(err).NotTo(o.HaveOccurred())
-					err = oc.Run("delete").Args("bc", "openshift-jee-sample").Execute()
+					_, err = j.GetJobConsoleLogsAndMatchViaBuildResult(br, "Finished: SUCCESS")
```

Comment:
by the way, this has the additional benefit of verifying the integration with oauth via the login plugin